crates/node/service/src/actors/engine/actor.rs

```diff
@@ -30,30 +30,44 @@ use crate::{NodeActor, actors::CancellableContext};
+    /// A channel to receive [`OpAttributesWithParent`] from the derivation actor.
+    attributes_rx: mpsc::Receiver<OpAttributesWithParent>,
+    /// A channel to receive [`OpExecutionPayloadEnvelope`] from the network actor.
+    unsafe_block_rx: mpsc::Receiver<OpExecutionPayloadEnvelope>,
+    /// A channel to receive reset requests.
     reset_request_rx: mpsc::Receiver<()>,
-    /// The sender for L2 safe head update notifications.
-    engine_l2_safe_head_tx: watch::Sender<L2BlockInfo>,
-    /// A channel to send a signal that EL sync has completed. Informs the derivation actor to
-    /// start. Because the EL sync state machine within [`InnerEngineState`] can only complete
-    /// once, this channel is consumed after the first successful send. Future cases where EL
-    /// sync is re-triggered can occur, but we will not block derivation on it.
-    sync_complete_tx: oneshot::Sender<()>,
-    /// A way for the engine actor to send a [`Signal`] back to the derivation actor.
-    derivation_signal_tx: mpsc::Sender<Signal>,
+    /// Handler for inbound queries to the engine.
+    inbound_queries: mpsc::Receiver<EngineQueries>,
+    /// A channel to receive [`RuntimeConfig`] from the runtime actor.
+    runtime_config_rx: mpsc::Receiver<RuntimeConfig>,
+    /// A channel to receive build requests from the sequencer actor.
+    /// TODO(@theochap): plug it in the engine actor.
+    #[allow(dead_code)]
+    build_request_rx:
+        mpsc::Receiver<(OpAttributesWithParent, mpsc::Sender<OpExecutionPayloadEnvelope>)>,
+    /// The [`L2Finalizer`], used to finalize L2 blocks.
+    finalizer: L2Finalizer,
 }
 
 /// The outbound data for the [`EngineActor`].
 #[derive(Debug)]
-pub struct EngineOutboundData {
-    /// A channel to request the engine actor to reset.
+pub struct EngineInboundData {
+    /// A channel to send [`OpAttributesWithParent`] and [`OpExecutionPayloadEnvelope`] to the
+    /// sequencer actor.
+    pub build_request_tx:
+        mpsc::Sender<(OpAttributesWithParent, mpsc::Sender<OpExecutionPayloadEnvelope>)>,
+    /// A channel to send [`OpAttributesWithParent`] to the derivation actor.
+    pub attributes_tx: mpsc::Sender<OpAttributesWithParent>,
+    /// A channel to send [`OpExecutionPayloadEnvelope`] to the network actor.
+    pub unsafe_block_tx: mpsc::Sender<OpExecutionPayloadEnvelope>,
+    /// A channel to send reset requests.
     pub reset_request_tx: mpsc::Sender<()>,
-    /// A channel to receive L2 safe head update notifications.
-    pub engine_l2_safe_head_rx: watch::Receiver<L2BlockInfo>,
-    /// A channel to receive a signal that EL sync has completed.
-    pub sync_complete_rx: oneshot::Receiver<()>,
-    /// A channel to send a [`Signal`] back to the derivation actor.
-    pub derivation_signal_rx: mpsc::Receiver<Signal>,
+    /// Handler to send inbound queries to the engine.
+    pub inbound_queries_tx: mpsc::Sender<EngineQueries>,
+    /// A channel to send [`RuntimeConfig`] to the runtime actor.
```

Comment:
This sender is given to the runtime actor to send to the engine actor iiuc

```suggestion
    /// A channel to send [`RuntimeConfig`] to the engine actor.
```